static/js/admin.js

```diff
@@ -163,7 +206,49 @@ exports.setup_page = function () {
         $("#deactivation_stream_modal .stream_name").text(stream_name);
         $("#deactivation_stream_modal").modal("show");
     });
-
+    $(".admin_default_stream_table").on("click", ".remove-default-stream", function (e) {
+        e.preventDefault();
+        e.stopPropagation();
+        $(".active_default_stream_row").removeClass("active_default_stream_row");
+        var row = $(e.target).closest(".default_stream_row");
+        row.addClass("active_default_stream_row");
+        var stream_name = row.find('.default_stream_name').text();
+        channel.del({
+            url: '/json/default_streams'+ '?' + $.param({"stream_name": stream_name}),
+            error: function (xhr, error_type) {
+                if (xhr.status.toString().charAt(0) === "4") {
+                    $(".active_default_stream_row button").closest("td").html(
+                    $("<p>").addClass("text-error").text($.parseJSON(xhr.responseText).msg));
+                } else {
+                    $(".active_default_stream_row button").text("Failed!");
+                }
+            },
+            success: function () {
+                var row = $(".active_default_stream_row");
+                row.remove();
+            }
+        });
+    });
+    $('.create_default_stream').typeahead({
+        items: 5,
+        fixed: true,
+        source: function (query,callback) {
+            channel.get({
+                url:      '/json/streams?include_public=true&include_subscribed=true&include_default=true',
+                timeout:  10*1000,
+                idempotent: true,
+                success: function (streams_data) {
+                    var non_default_streams_names = get_non_default_streams_names(streams_data);
+                    callback(non_default_streams_names);
+                },
+                error: failed_listing_streams
+            });
```

Comment:
Hmm, don't we already have the data for which streams exist and are or are not default streams from the `populate_streams` table (and actually, probably also in the data sent from the server in subs.js?).  It would be great if we could just reuse the data we're already fetching rather than adding another AJAX query for the server here.